SonarQube.TeamBuild.PostProcessor/SummaryReportBuilder.cs

```diff
@@ -0,0 +1,182 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="SummaryReportBuilder.cs" company="SonarSource SA and Microsoft Corporation">
+//   Copyright (c) SonarSource SA and Microsoft Corporation.  All rights reserved.
+//   Licensed under the MIT License. See License.txt in the project root for license information.
+// </copyright>
+//-----------------------------------------------------------------------
+using SonarQube.Common;
+using SonarQube.TeamBuild.Integration;
+using SonarRunner.Shim;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+
+namespace SonarQube.TeamBuild.PostProcessor
+{
+    /// <summary>
+    /// Generates summary reports for various build systems
+    /// </summary>
+    internal class SummaryReportBuilder
+    {
+        public class SummaryReportData
```

Comment:
This probably doesn't need to be public (it would have done had we been plugging in different types of report generator).